### PR TITLE
fix: Added IgnoreError filter for diagnostic setting resource

### PR DESCRIPTION
**Pull request by amanenk**
🎉 Thank you for making CloudQuery awesome by submitting a PR 🎉

#### Summary

fixed unreachable code in errorClassifier
fixes error when `SubscriptionNotRegistered` for resource usage
closes https://sentry.io/organizations/cloudquery/issues/3297729651/?query=is%3Aunresolved+microsoft.insights&statsPeriod=14d and similar issues

---

Use the following steps to ensure your PR is ready to be reviewed

- [x] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [x] Run `go fmt` to format your code 🖊
- [x] Lint your changes via `golangci-lint run --new-from-rev main` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [x] Update or add tests. Learn more about testing [here](https://docs.cloudquery.io/docs/developers/sdk/testing) 🧪
- [x] Update the docs by running `go run ./docs/docs.go` and committing the changes 📃
- [x] If adding a new resource, add [relevant Terraform files](../terraform) in a separate PR 📂
- [x] Ensure the status checks below are successful ✅

### CODE REVIEW

**Comment by amanenk** [comment]
I can't reproduce it even on accounts where `microsoft.insights` is disabled but the fix should help to avoid this error

File: client/helpers.go [hunk 1]
```diff
@@ -61,3 +63,11 @@ func IgnoreAccessDenied(err error) bool {
...
 	return errors.As(err, &detailedError) && detailedError.StatusCode == http.StatusForbidden
 }
+
+func IgnoreSubscriptionNotRegistered(err error) bool {
+	var azureErr *azure.RequestError
+	if errors.As(err, &azureErr) {
+		return azureErr.ServiceError != nil && azureErr.ServiceError.Code == "SubscriptionNotRegistered"
+	}
+	return false
+}
```
Comment (bbernays) [inline client/helpers.go:73]: Why not add this to the generic error classifier that way all resources can benefit from it?

Comment (amanenk) [reply client/helpers.go]: moved to the generic error classifier and fixed unreachable code there.

**Review by erezrokah — APPROVED** [review]
🚀

**Pull request merged**